ngraph/frontend/paddlepaddle/src/pdpd_fw_node.cpp

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <pdpd_fw_node.hpp>
+
+namespace ngraph
+{
+    namespace frontend
+    {
+        NGRAPH_RTTI_DEFINITION(PDPDFrameworkNode, "PDPDFrameworkNode", 1);
+
+        std::map<std::string, OutputVector> PDPDFrameworkNode::get_named_inputs() const
+        {
+            return m_decoder.map_for_each_input([&](std::string name) {
+                auto it = std::find(m_inputs_names.begin(), m_inputs_names.end(), name);
+                if (it != m_inputs_names.end())
+                {
+                    return input(it - m_inputs_names.begin()).get_source_output();
+                }
+                else
+                {
+                    return Output<Node>();
+                }
+            });
+        }
+
+        std::map<std::string, OutputVector> PDPDFrameworkNode::get_named_outputs()
+        {
+            size_t idx = 0;
+            return m_decoder.map_for_each_output([&](std::string name) { return output(idx++); });
```

Comment:
should be 'const std::string &'

Reply:
Is `name` really not needed here?

Reply:
`idx `should be always 0?

Reply:
> idx should be always 0?
Yes, it starts from 0.

Reply:
> Is name really not needed here?

Yes it is not needed here, but I decided to make it in a similar way as inputs.